src/addrman.cpp

```diff
@@ -731,13 +736,18 @@ CAddrInfo CAddrMan::Select_(bool newOnly) const
         // use a new node
         double fChanceFactor = 1.0;
         while (1) {
+            // Pick a new bucket, and an initial position in that bucket.
             int nUBucket = insecure_rand.randrange(ADDRMAN_NEW_BUCKET_COUNT);
             int nUBucketPos = insecure_rand.randrange(ADDRMAN_BUCKET_SIZE);
-            while (vvNew[nUBucket][nUBucketPos] == -1) {
-                nUBucket = (nUBucket + insecure_rand.randbits(ADDRMAN_NEW_BUCKET_COUNT_LOG2)) % ADDRMAN_NEW_BUCKET_COUNT;
-                nUBucketPos = (nUBucketPos + insecure_rand.randbits(ADDRMAN_BUCKET_SIZE_LOG2)) % ADDRMAN_BUCKET_SIZE;
+            // Iterate over the positions of that bucket.
+            int i;
+            for (i = 0; i < ADDRMAN_BUCKET_SIZE; ++i) {
+                if (vvNew[nUBucket][nUBucketPos ^ i] != -1) break;
```

Comment:
nit: I think it's not particularly intuitive that this loop with `nUBucketPos ^ i` exhausts all values [0; ADDRMAN_BUCKET_SIZE). Perhaps we could use something more explicit/readable?

Reply:
Made some changes; perhaps it's clearer now.

Reply:
Hmmm, maybe just personal taste, but I think the old version is better. Having a separate counter (`i`) from the thing being incremented (`nUBucketPos`) seems unusual and a bit jarring. Both work, but I have a mild preference for the original.